syncapi/consumers/clientapi.go

```diff
@@ -92,7 +96,8 @@ func (s *OutputClientDataConsumer) onMessage(msg *sarama.ConsumerMessage) error
 		}).Panicf("could not save account data")
 	}
 
-	s.notifier.OnNewEvent(nil, "", []string{string(msg.Key)}, types.StreamingToken{PDUPosition: pduPos})
+	s.streams.AccountDataStreamProvider.Advance(streamPos)
```

Comment:
It feels odd that writes to create new positions are done here in the consumer but recording this is in the provider. It feels poorly encapsulated, as I would expect a stream provider to be able to handle insertion too.

Reply:
Indeed, but given that the `StreamProvider` and `PartitionedStreamProvider` interfaces have to be generic, I didn't really know how best to handle inserts and didn't give it a lot of thought. Happy to revisit this in the next PR.